academic_observatory_workflows/workflows/pubmed_telescope.py

```diff
@@ -0,0 +1,1345 @@
+                    with open(download_location, "rb") as f_hash:
+                        data = f_hash.read()
+                        md5hash_from_download = hashlib.md5(data).hexdigest()
+
+                    logging.info(f"MD5 has from downloaded file: {md5hash_from_download}")
+
+                    # Need to have a download catch for the hash file as well, otherwise it can break the download loop.
+                    try:
+                        # Download corresponding md5 hash.
+                        with open(f"{download_location}.md5", "wb") as f:
+                            ftp_conn.retrbinary(f"RETR {changefile.path_on_ftp}.md5", f.write)
+                    except:
+                        logging.info(
+                            f"Unable to download {changefile.path_on_ftp}.md5 from PubMed's FTP server {self.ftp_server_url}."
+                        )
+
+                    # Peep into md5 file.
+                    with open(f"{download_location}.md5", "r") as f_md5:
+                        md5_from_pubmed_ftp = f_md5.read()
+
+                    # If md5 does not match, raise an Airflow exception.
+                    if md5hash_from_download in md5_from_pubmed_ftp:
+                        download_success = True
+                    else:
+                        logging.info(
+                            f"MD5 hash does not match the given MD5 checksum from server: {changefile.download_file_path}"
+                        )
+
+                    download_attemp_count += 1
+
+                if not download_success:
+                    raise AirflowException(
+                        f"Unable to download {download_location} from PubMed's FTP server {self.ftp_server_url} after {self.max_download_retry}"
+                    )
+
+            else:
+                logging.info(f"Downloading: {changefile.download_file_path}")
+                try:
+                    # Download file
+                    with open(changefile.download_file_path, "wb") as f:
+                        ftp_conn.retrbinary(f"RETR {changefile.path_on_ftp}", f.write)
+                except:
+                    raise AirflowException(
+                        f"Unable to download {changefile.path_on_ftp} from PubMed's FTP server {self.ftp_server_url}"
+                    )
+
+        # Close the FTP connection after downloading the required files.
+        ftp_conn.close()
+
+    def upload_downloaded(self, release: PubMedRelease, **kwargs):
+        """
+        Push all downloaded changefiles from Pubmed to Google Cloud Storage.
+        """
+
+        # Grab list of files to upload.
+        changefiles_to_upload = [changefile.download_file_path for changefile in release.changefile_list]
+
+        success = gcs_upload_files(
+            bucket_name=self.cloud_workspace.download_bucket,
+            file_paths=changefiles_to_upload,
+        )
+
+        if not success:
```

Comment:
Jamie made a function during the observatory platform's recent refactor that's built to streamline success checks. See here for an example: 
https://github.com/The-Academic-Observatory/oaebu-workflows/blob/30e3ed251e7d348d29ad1f774ade6f08bde0fc96/oaebu_workflows/workflows/jstor_telescope.py#L293